### PR TITLE
feat(curriculum): add English curriculum block 5.2 describing code-related concepts and terms

### PR DESCRIPTION
**Pull request by miyaliu666**
Checklist:

- [x] I have read and followed the [contribution guidelines](https://contribute.freecodecamp.org).
- [x] I have read and followed the [how to open a pull request guide](https://contribute.freecodecamp.org/#/how-to-open-a-pull-request).
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

### CODE REVIEW

File: curriculum/challenges/english/21-a2-english-for-developers/learn-how-to-use-code-related-concepts-and-terms/6633032bf786ff8a61b0c482.md [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+
+The phrase `thanks for` should not be followed by `to` + verb. It should be a gerund or a noun.
+
+---
+
+`Thanks for assisting with the task.`
+
+---
+
+`Thanks for the assist with the task.`
```
Comment (naomi-lgbt) [inline curriculum/challenges/english/21-a2-english-for-developers/learn-how-to-use-code-related-concepts-and-terms/6633032bf786ff8a61b0c482.md:40]: This is actually grammatically correct. I wonder if we should use a different example?

File: curriculum/challenges/english/21-a2-english-for-developers/learn-how-to-use-code-related-concepts-and-terms/663354f60aafd69a4c0138da.md [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+Sophia: You can use the equals symbol to assign values to them.
+-->
+
+# --description--
+
+The `equals symbol` (`=`) is commonly used for assignment. It's used to assign a value to a variable. The variable name is written on the left of the equals symbol, and the value you want to store in the variable is on the right.
+
+For example, `name = "Sophia";` assigns the string `"Sophia"` to the variable `name`.
+
+To `assign value` in programming means to give a specific value to a variable. The `equals symbol` (`=`) is commonly used for this purpose. When you assign a value to a variable, you are telling the program to remember that value for use later.
```
Comment (naomi-lgbt) [inline curriculum/challenges/english/21-a2-english-for-developers/learn-how-to-use-code-related-concepts-and-terms/663354f60aafd69a4c0138da.md:19]:
```suggestion
To `assign a value` in programming means to give a specific value to a variable. The `equals symbol` (`=`) is commonly used for this purpose. When you assign a value to a variable, you are telling the program to remember that value for use later.
```

File: curriculum/challenges/_meta/learn-how-to-use-code-related-concepts-and-terms/meta.json [hunk 3]
```diff
@@ -9,8 +9,216 @@
   "superBlock": "a2-english-for-developers",
   "challengeOrder": [
     {
-      "id": "655c12384a3d2e8a0d5e3efd",
-      "title": "Dialogue: Placeholder"
+      "id": "6630b37556424464ba0f984e",
+      "title": "Dialogue 1: Asking about data types"
```
Comment (moT01) [inline curriculum/challenges/_meta/learn-how-to-use-code-related-concepts-and-terms/meta.json:13]:
```suggestion
      "title": "Dialogue 1: Asking about Data Types"
```

File: curriculum/challenges/_meta/learn-how-to-use-code-related-concepts-and-terms/meta.json [hunk 4]
```diff
@@ -9,8 +9,216 @@
...
+      "id": "6633203e58595e93ef54ba3b",
+      "title": "Task 33"
+    },
+    {
+      "id": "6633261bb572f2953f5abd13",
+      "title": "Task 34"
+    },
+    {
+      "id": "66334fd916a3a697e1d2c631",
+      "title": "Dialogue 3: Getting help with variables"
```
Comment (moT01) [inline curriculum/challenges/_meta/learn-how-to-use-code-related-concepts-and-terms/meta.json:157]:
```suggestion
      "title": "Dialogue 3: Getting Help with Variables"
```

File: curriculum/challenges/english/21-a2-english-for-developers/learn-how-to-use-code-related-concepts-and-terms/66334fd916a3a697e1d2c631.md [hunk 5]
```diff
@@ -0,0 +1,14 @@
+---
+id: 66334fd916a3a697e1d2c631
+title: "Dialogue 3: Getting help with variables"
```
Comment (moT01) [inline curriculum/challenges/english/21-a2-english-for-developers/learn-how-to-use-code-related-concepts-and-terms/66334fd916a3a697e1d2c631.md:3]:
```suggestion
title: "Dialogue 3: Getting Help with Variables"
```

File: curriculum/challenges/english/21-a2-english-for-developers/learn-how-to-use-code-related-concepts-and-terms/6633ad387b6914a3313339b3.md [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+
+`Touch` often means to come into physical contact with something. It can also be used metaphorically in contexts like programming, where it means to alter or affect something.
+
+`These` is used to refer to specific items that are close to the speaker or are being actively discussed.
+
+# --fillInTheBlank--
+
+## --sentence--
+
+`I see. _, that means I _ change the values in _ variables, but I _ _ the values in these constants.`
```
Comment (naomi-lgbt) [inline curriculum/challenges/english/21-a2-english-for-developers/learn-how-to-use-code-related-concepts-and-terms/6633ad387b6914a3313339b3.md:23]:
```suggestion
`I see. _ that means I _ change the values in _ variables, but I _ _ the values in these constants.`
```

**Review by naomi-lgbt — CHANGES_REQUESTED** [review]

File: curriculum/challenges/english/21-a2-english-for-developers/learn-how-to-use-code-related-concepts-and-terms/6630bd815e1b146937422948.md [hunk 7]
```diff
@@ -0,0 +1,65 @@
...
+
+For example, `At the restaurant, you can choose from ten different desserts.` In this sentence, `choose from` is used to imply that the restaurant offers a variety of ten desserts, and you have the freedom to make a selection based on your preference.
+
+`Which one to choose` is a common phrase used when someone needs to decide between multiple options and is unsure which is the best choice.
+
+# --fillInTheBlank--
+
+## --sentence--
+
+`Jake: Thanks. But I see _ data types listed in the documentation. How do I know _ _ _ _?`
```
Comment (moT01) [inline curriculum/challenges/english/21-a2-english-for-developers/learn-how-to-use-code-related-concepts-and-terms/6630bd815e1b146937422948.md:25]:
```suggestion
`Thanks. But I see _ data types listed in the documentation. How do I know _ _ _ _?`
```

**Review by moT01 — CHANGES_REQUESTED** [review]
Looking good @miyaliu666 🎉 Found a few minor things...

File: curriculum/challenges/english/21-a2-english-for-developers/learn-how-to-use-code-related-concepts-and-terms/6633ad387b6914a3313339b3.md (hunk 6 above)
Comment (miyaliu666) [reply curriculum/challenges/english/21-a2-english-for-developers/learn-how-to-use-code-related-concepts-and-terms/6633ad387b6914a3313339b3.md:23]: There is a comma after the word "So".

File: curriculum/challenges/english/21-a2-english-for-developers/learn-how-to-use-code-related-concepts-and-terms/6633032bf786ff8a61b0c482.md (hunk 1 above)
Comment (miyaliu666) [reply curriculum/challenges/english/21-a2-english-for-developers/learn-how-to-use-code-related-concepts-and-terms/6633032bf786ff8a61b0c482.md]: I remove "the" from the sentence, so the feedback makes more sense.

**Review by naomi-lgbt — APPROVED** [review]

**Review by moT01 — APPROVED** [review]

**Pull request merged**